docs/markdown/Contributions/development/style-guide.md

```diff
@@ -389,6 +389,37 @@ merge_constraints([constraint for constraint in all_constraints if constraint.st
 
 The return type, however, should usually be as precise as possible so that call sites have better type inference.
 
+### Use descriptive names for `TypeVar`s
```

Comment:
We might want typevars to start starting with underscores so they aren't assumed to be public.

Reply:
I don't think _requiring_ typevars to be private is a good idea. A very rough e.g: if you're making a generic subclass in a different model, you'd probably want to re-use the typecar from the superclass.

Reply:
_encourage_ then? :smile: 

My editor has tried to suggest TypeVars from other modules before. Super annoying.

Reply:
Fine :)